executor/join.go

```diff
@@ -294,6 +294,10 @@ func (e *HashJoinExec) fetchOuterChunks(goCtx goctx.Context) {
 			close(e.outerResultChs[i])
 		}
 		e.workerWaitGroup.Done()
+
+		if r := recover(); r != nil {
```

Comment:
This should before `e.workerWaitGroup.Done()`.
When `WaitGroup.Done()` is called, the `e.joinResultCh` maybe close immediately, and send to a closed channel will panic.